src/TokenizerDelim.cpp

```diff
@@ -160,9 +161,17 @@ Token TokenizerDelim::nextToken() {
 
     case STATE_COMMENT:
       if (*cur_ == '\r' || *cur_ == '\n') {
+
+        // If we have read at least one record on the current row go to the
+        // next row, line, otherwise just ignore the line.
+        if (col_ > 0) {
+          newRecord();
+          ++row;
```

Comment:
That is true, but then you get all NAs for commented out lines in the output, which differs from the current convention (and that of `read.csv()`).